src/engine_openssl.c

```diff
@@ -295,6 +305,9 @@ static int tls_verify_signature(void *cert, enum hash_algo md, const char* data,
 
 static void tls_free_ctx(tls_context *ctx) {
     struct openssl_ctx *c = ctx->ctx;
+    if (c->alpn_protocols) {
+        free(c->alpn_protocols);
```

Comment:
shouldn't this free each entry like how `mbedtls_set_alpn_protocols` does? also - why not make this a function and stay DRY? same comment below in `tls_set_alpn_protocols`

Reply:
openssl uses different structure to configure ALPN